src/widgets/heightandweight/heightandweight-overview.component.tsx

```diff
@@ -37,62 +40,91 @@ export default function HeightAndWeightOverview(
-                <th></th>
-              </tr>
-            </thead>
-            <tbody>
-              {dimensions.slice(0, showMore ? 6 : 3).map((dimension, index) => (
-                <tr key={dimension.id} className={styles.tableRow}>
-                  <td
-                    className={styles.tableData}
-                    style={{ textAlign: "start" }}
-                  >
-                    {dimension.date}
-                  </td>
-                  <td className={styles.tableData}>
-                    {dimension.weight || "\u2014"}
-                    <span className={styles.unit}>{index === 0 && " kg"}</span>
-                  </td>
-                  <td className={styles.tableData}>
-                    {dimension.height || "\u2014"}
-                    <span className={styles.unit}>{index === 0 && " cm"}</span>
-                  </td>
-                  <td className={styles.tableData}>
-                    {dimension.bmi || "\u2014"}
-                    {}
-                    <span className={styles.unit}>
-                      {index === 0 && " kg/m"}
-                      {index === 0 && <sup>2</sup>}
-                    </span>
-                  </td>
-                  <td style={{ textAlign: "end" }}>
-                    <Link to={`${heightweightPath}/${dimension.id}`}>
-                      <svg className="omrs-icon" fill="rgba(0, 0, 0, 0.54)">
-                        <use xlinkHref="#omrs-icon-chevron-right" />
-                      </svg>
-                    </Link>
-                  </td>
-                </tr>
-              ))}
-            </tbody>
-          </table>
-        </SummaryCardRowContent>
-      </SummaryCardRow>
-      <SummaryCardFooter linkTo={`${heightweightPath}`} />
-    </SummaryCard>
+    <>
+      {dimensions.length > 0 ? (
+        <SummaryCard
+          name="Height & Weight"
+          link={`${heightweightPath}`}
+          showComponent={() =>
+            openHeightAndWeightTab(VitalsForm, "Vitals Form")
+          }
+          addComponent={VitalsForm}
+        >
+          <SummaryCardRow>
+            <SummaryCardRowContent>
+              <table className={styles.table}>
+                <thead>
+                  <tr className={styles.tableRow}>
+                    <th
+                      className={`${styles.tableHeader} ${styles.tableDates}`}
+                      style={{ textAlign: "start" }}
+                    ></th>
+                    <th className={styles.tableHeader}>Weight</th>
+                    <th className={styles.tableHeader}>Height</th>
+                    <th className={styles.tableHeader}>BMI</th>
+                    <th></th>
+                  </tr>
+                </thead>
+                <tbody>
+                  {dimensions
+                    .slice(0, showMore ? 6 : 3)
+                    .map((dimension, index) => (
+                      <tr key={dimension.id} className={styles.tableRow}>
```

Comment:
Hehe, I'm always looking for missing `key` props - but I never find them! This is what I call careful coding. Great stuff! 🚀